updated dependencies + ugly build fix

antonkulaga opened the pull request:
At the moment banana-rdf is not usable as both Intellij and Eclipse imports are broken. Here I updated some of the dependencies and introduced dirty fixes that make it possible to import the project to IDEs. @InTheNow will be very busy during next several weeks so I suggest to use dirty fix as a temporal solution before he will find some time to fist the bug in sbt-scalajs plugin ( I opened an issue about this https://github.com/InTheNow/sbt-scalajs/issues/17 )

antonkulaga committed:
updated dependencies + movement to CommonBuild to fix the build issue

bblfish commented:
seems good enough to me this "ugly" fix. Given that it is beta, and that this is an improvement over what we have I suggest we merge it. Also there are some good improvements such as moving to latest scala-js.

antonkulaga commented:
@bblfish 

I deleted ugly parts and I moved to CommonBuild. I borrowed it from https://github.com/w3c/banana-rdf/tree/fix/sbt-scalajs-0.6.0 but I decided not to push there as it was harder ( that branch was 18 commits behind) than changing my PR.

bblfish commented:
can I merge this?

antonkulaga commented:
I will be happy to see it merged of course if there are no objections from @InTheNow and @betehess

betehess commented on sesame/src/main/scala/org/w3/banana/sesame/SesameOps.scala:
```diff
@@ -167,7 +167,7 @@ class SesameOps extends RDFOps[Sesame] with SesameMGraphOps with DefaultURIOps[S
     val leftNoContext = left.asScala.map(s => makeTriple(s.getSubject, s.getPredicate, s.getObject)).asJava
     val rightNoContext = right.asScala.map(s => makeTriple(s.getSubject, s.getPredicate, s.getObject)).asJava
-    ModelUtil.equals(leftNoContext, rightNoContext)
+    Models.isomorphic(leftNoContext, rightNoContext)
```
what motivates that change?

antonkulaga replied:
I moved from outdated beta version of Sesame to newer (and stable). And in stable one there is not ModuleUtil has been deprecated and Models.isomorphic is recommended instead of ModelUtil.equals

betehess replied:
Ok. So in that case, is the workaround still needed? Is the comment still accurate? Can you please check?

This is _pretty important_ as this touches the core implementation of the RDFOps typeclass, which is supposed to be as close as possible to the native implementation.

antonkulaga replied:
Well, all tests pass for sesame. An update from 2.8 beta to stable 2.8.x (that I made here) is a thing we cannot avoid, and it is not possible to use ModelUtil.equals in 2.8.x stable

bblfish replied:
@antonkulaga I think @betehess is asking if you checked that the two lines above actually are still correct independent of our tests that may not be complete, since those lines were introduced to fix an issue as explained in the javadoc above

``` java
  /**
   * the new ModelUtil.equals changed its semantics. See
   * - https://openrdf.atlassian.net/browse/SES-1695
   * - https://groups.google.com/forum/#!topic/sesame-devel/CGFDn7mESLg/discussion
   */
```

InTheNow commented:
Some, but not all, changes from https://github.com/w3c/banana-rdf/tree/fix/build-scalajs seem to have been manually added here - is that right? Because it has quite hard to see

antonkulaga commented:
@InTheNow , yes I manually added all relevant changed from fix/build-scalajs that I saw. The fix/build-scalajs branch also was contained some things (like disabling N3) that are redundant now, of course I did not add them.

antonkulaga commented:
@InTheNow asked me to rebase and squash on top of fix_scalajs_branch. I did this as an another PR ( https://github.com/w3c/banana-rdf/pull/264 )

P.S. I close this PR to avoid confusion. The PR about Sesame will come after #264 will be merged
